near-sdk/src/types/public_key.rs

```diff
@@ -0,0 +1,282 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use bs58::decode::Error as B58Error;
+use std::convert::TryFrom;
+
+/// PublicKey curve
+#[derive(Debug, Clone, Copy, PartialOrd, Ord, Eq, PartialEq, BorshDeserialize, BorshSerialize)]
+#[repr(u8)]
+pub enum CurveType {
+    ED25519 = 0,
+    SECP256K1 = 1,
+}
+
+impl CurveType {
+    fn from_u8(val: u8) -> Result<Self, ParsePublicKeyError> {
+        match val {
+            0 => Ok(CurveType::ED25519),
+            1 => Ok(CurveType::SECP256K1),
+            _ => Err(ParsePublicKeyError { kind: ParsePublicKeyErrorKind::UnknownCurve }),
+        }
+    }
+}
+
+impl TryFrom<String> for CurveType {
+    type Error = ParsePublicKeyError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        Ok(value.parse::<Self>()?)
+    }
+}
```

Comment:
I don't think we need this?

Reply:
Yeah, this shouldn't be needed. Also shouldn't break anything since CurveType was never exposed before